src/components/Menu/Menu.tsx

```diff
@@ -49,33 +55,60 @@ export type MenuItemsProps = ExtractProps<typeof HeadlessMenu.Items>;
  *
  * A dropdown that reveals or hides a list of actions
  */
-export const Menu = ({ className, ...other }: MenuProps) => {
+export const Menu = ({
+  buttonText,
+  children,
+  className,
+  ...other
+}: MenuProps) => {
   const menuClassNames = clsx(className, styles['menu']);
-  return <HeadlessMenu as="div" className={menuClassNames} {...other} />;
+  if (typeof children === 'function') {
+    return (
+      <HeadlessMenu as="div" className={menuClassNames} {...other}>
+        {({ open, close }) => (
+          <>
+            {buttonText && (
+              <MenuButton className={styles['menu__button']}>
+                {buttonText}
+                <Icon
+                  className={styles['menu__button--icon']}
+                  name="expand-more"
+                  purpose="decorative"
+                  size="1.25rem"
+                />
+              </MenuButton>
+            )}
+            {children({ open, close })}
+          </>
+        )}
+      </HeadlessMenu>
+    );
+  }
+  return (
+    <HeadlessMenu as="div" className={menuClassNames} {...other}>
+      {buttonText && (
+        <MenuButton className={styles['menu__button']}>
+          {buttonText}
+          <Icon
+            className={styles['menu__button--icon']}
+            name="expand-more"
+            purpose="decorative"
+            size="1.25rem"
+          />
+        </MenuButton>
+      )}
+      {children}
+    </HeadlessMenu>
+  );
 };
 
 /**
  * A button that when clicked, shows or hides the Options
+ * Refer to Button documentation for more style variants.
  */
-const MenuButton = ({ children, className, ...other }: MenuButtonProps) => {
-  const buttonClassNames = clsx(styles['menu__button'], className);
+const MenuButton = ({ status, ...other }: MenuButtonProps) => {
   return (
-    <HeadlessMenu.Button
-      as={ClickableStyle}
-      className={buttonClassNames}
-      status="neutral"
-      {...other}
-    >
-      <>
-        {children}
-        <Icon
-          className={styles['menu__button--icon']}
-          name="expand-more"
-          purpose="decorative"
-          size="1.25rem"
-        />
-      </>
-    </HeadlessMenu.Button>
+    <HeadlessMenu.Button as={Button} status={status || 'neutral'} {...other} />
```

Comment:
Kept `as={Button}` for accessibility features (such as target size) and EDS variants

Reply:
Dang, Headless's Menu.Button isn't necessarily a button? 😬

Reply:
It is but I was thinking keeping it EDS themed hence the EDS Button